prototypes/matrix_free_navier_stokes/matrix_free_navier_stokes.cc

```diff
@@ -0,0 +1,2420 @@
+  MGLevelObject<std::unique_ptr<OperatorType>>       operators;
+  MGLevelObject<MGTwoLevelTransfer<dim, VectorType>> transfers;
+  MGLevelObject<VectorType>                          mg_solution;
+
+  std::vector<std::shared_ptr<const Triangulation<dim>>>
+    coarse_grid_triangulations;
+
+  coarse_grid_triangulations =
+    MGTransferGlobalCoarseningTools::create_geometric_coarsening_sequence(
+      dof_handler.get_triangulation());
+
+  const unsigned int n_h_levels = coarse_grid_triangulations.size();
+
+  unsigned int minlevel = 0;
+  unsigned int maxlevel = n_h_levels - 1;
+
+  dof_handlers.resize(minlevel, maxlevel);
+  operators.resize(minlevel, maxlevel);
+  transfers.resize(minlevel, maxlevel);
+  mg_solution.resize(minlevel, maxlevel);
+
+  for (unsigned int l = minlevel; l <= maxlevel; ++l)
+    {
+      dof_handlers[l].reinit(*coarse_grid_triangulations[l]);
+      dof_handlers[l].distribute_dofs(dof_handler.get_fe());
+    }
+
+  MGLevelObject<AffineConstraints<typename VectorType::value_type>> constraints(
+    minlevel, maxlevel);
+
+  for (unsigned int level = minlevel; level <= maxlevel; ++level)
+    {
+      const auto &dof_handler = dof_handlers[level];
+      auto &      constraint  = constraints[level];
+
+      const IndexSet locally_relevant_dofs =
+        DoFTools::extract_locally_relevant_dofs(dof_handler);
+      constraint.reinit(locally_relevant_dofs);
+
+      DoFTools::make_hanging_node_constraints(dof_handler, constraint);
+      VectorTools::interpolate_boundary_values(mapping,
+                                               dof_handler,
+                                               0,
+                                               Functions::ZeroFunction<dim>(
+                                                 dim + 1),
+                                               constraint);
+      VectorTools::interpolate_boundary_values(mapping,
+                                               dof_handler,
+                                               1,
+                                               Functions::ZeroFunction<dim>(
+                                                 dim + 1),
+                                               constraint);
+      VectorTools::interpolate_boundary_values(mapping,
+                                               dof_handler,
+                                               2,
+                                               Functions::ZeroFunction<dim>(
+                                                 dim + 1),
+                                               constraint);
+      VectorTools::interpolate_boundary_values(mapping,
+                                               dof_handler,
+                                               3,
+                                               Functions::ZeroFunction<dim>(
+                                                 dim + 1),
+                                               constraint);
+      if constexpr (dim == 3)
```

Comment:
Does the stokes example not support 3D?

Reply:
It does support 3D, but I probably didn't add the MMS. I will add it in a quick PR so that it is also complete.